### PR TITLE
Psr 1238

### PR DESCRIPTION
**Pull request by narendrapaduchuri**

### CODE REVIEW

File: app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala [hunk 1]
```diff
@@ -265,12 +312,32 @@ object SharesDisposalCYAController {
...
         .withMarginBottom(Margin.Fixed60Bottom),
       refresh = None,
-      buttonText = parameters.mode.fold(
-        normal = "site.saveAndContinue",
-        check = "site.continue"
-      ),
-      onSubmit = routes.SharesDisposalCYAController
-        .onSubmit(parameters.srn, parameters.shareIndex, parameters.disposalIndex, parameters.mode)
+      buttonText =
+        parameters.mode.fold(normal = "site.saveAndContinue", check = "site.continue", viewOnly = "site.continue"),
```
Comment (kberec) [inline app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala:316]: should viewOnly be `site.return.to.tasklist` ?

File: app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala [hunk 2]
```diff
@@ -265,12 +312,32 @@ object SharesDisposalCYAController {
...
+        .onSubmit(parameters.srn, parameters.shareIndex, parameters.disposalIndex, parameters.mode),
+      optViewOnlyDetails = if (mode == ViewOnlyMode) {
+        Some(
+          ViewOnlyDetailsViewModel(
+            updated = viewOnlyUpdated,
+            link = None,
+            submittedText = Some(Message("")),
+            title = "sharesDisposal.viewOnly.title",
+            heading = Message("sharesDisposal.viewOnly.heading", parameters.companyName),
+            buttonText = "site.continue",
```
Comment (kberec) [inline app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala:327]: should this be site.return.to.tasklist ?

File: test/controllers/nonsipp/sharesdisposal/SharesDisposalCYAControllerSpec.scala [hunk 3]
```diff
@@ -292,4 +321,86 @@ class SharesDisposalCYAControllerSpec extends ControllerBaseSpec {
...
+              ViewOnlyMode,
+              viewOnlyUpdated = false,
+              optYear = Some(yearString),
+              optCurrentVersion = Some(submissionNumberTwo),
+              optPreviousVersion = Some(submissionNumberOne),
+              compilationOrSubmissionDate = Some(submissionDateTwo)
+            )
+          )
+      }
+    )
```
Comment (kberec) [inline test/controllers/nonsipp/sharesdisposal/SharesDisposalCYAControllerSpec.scala:393]: Does this need a`.withName`

File: app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala (hunk 1 above)
Comment (narendrapaduchuri) [reply app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala:316]: Hi @kberec ,  QA's telling that i should go to previous page not the task list page from CYA, and in prototype also behaving the same. It should be 'site.continue'

File: app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala (hunk 2 above)
Comment (narendrapaduchuri) [reply app/controllers/nonsipp/sharesdisposal/SharesDisposalCYAController.scala:327]: Here also same @kberec

**Review by kberec — APPROVED** [review]

**Review by kberec — APPROVED** [review]

**Pull request merged**